sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java

```diff
@@ -505,7 +506,8 @@ private static TelemetryItemExporter createTelemetryItemExporter(HttpPipeline ht
                         telemetryPipeline,
                         // TODO (trask) change this to statsbeatModule.getNonessentialStatsbeat()?
                         LocalStorageStats.noop(),
-                        false));
+                        false),
+                    new StatsbeatTelemetryPipelineListener(statsbeatModule::shutdown));
```

Comment:
I think we only want to register this listener on the statsbeat telemetry pipeline, I sent #37611 to split out a separate statsbeat telemetry pipeline which is what we do in the agent